lib/auth/accesslist/service_test.go

```diff
@@ -0,0 +1,256 @@
+// Copyright 2023 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package accesslist
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/stretchr/testify/require"
+
+	conv "github.com/gravitational/teleport/api/convert/teleport/accesslist/v1"
+	accesslistv1 "github.com/gravitational/teleport/api/gen/proto/go/teleport/accesslist/v1"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/api/types/header"
+	"github.com/gravitational/teleport/lib/authz"
+	"github.com/gravitational/teleport/lib/backend/memory"
+	"github.com/gravitational/teleport/lib/services"
+	"github.com/gravitational/teleport/lib/services/local"
+	"github.com/gravitational/teleport/lib/tlsca"
+)
+
+func TestAccessLists(t *testing.T) {
+	ctx, svc := initSvc(t)
+
+	getResp, err := svc.GetAccessLists(ctx, &accesslistv1.GetAccessListsRequest{})
+	require.NoError(t, err)
+	require.Empty(t, getResp.AccessLists)
+
+	a1 := newAccessList(t, "1")
+	a2 := newAccessList(t, "2")
+	a3 := newAccessList(t, "3")
+
+	_, err = svc.UpsertAccessList(ctx, &accesslistv1.UpsertAccessListRequest{AccessList: conv.ToV1(a1)})
+	require.NoError(t, err)
+
+	_, err = svc.UpsertAccessList(ctx, &accesslistv1.UpsertAccessListRequest{AccessList: conv.ToV1(a2)})
+	require.NoError(t, err)
+
+	_, err = svc.UpsertAccessList(ctx, &accesslistv1.UpsertAccessListRequest{AccessList: conv.ToV1(a3)})
+	require.NoError(t, err)
+
+	getResp, err = svc.GetAccessLists(ctx, &accesslistv1.GetAccessListsRequest{})
+	require.NoError(t, err)
+	require.Empty(t, cmp.Diff([]*types.AccessList{a1, a2, a3}, mustFromV1All(t, getResp.AccessLists...),
```

Comment:
Do we expect results to always be sorted by name? Could we say that explicitly?

Reply:
Hrm, in this case I think they're just in the order that they're inserted in? Though that happens to be alphabetical here. I don't have a strong need for whatever order is present here, however.

Reply:
If we don't care about the order, then let's reflect that in the assertion. It's good to be clear about the tests' intent, and also it could flake if the result order changes.

I don't suppose we care much about order in GetAll RPCs, but it may be best to state any guarantees (or lack thereof) explicit in the RPC too.

Reply:
Recently I found out about this: https://pkg.go.dev/github.com/google/go-cmp/cmp/cmpopts#SortSlices. It helps when doing order-independent comparisons.

Reply:
Added in this SortSlices bit!